src/app/fdctl/run/tiles/fd_net.c

```diff
@@ -355,6 +476,8 @@ unprivileged_init( fd_topo_t *      topo,
   ulong scratch_top = FD_SCRATCH_ALLOC_FINI( l, 1UL );
   if( FD_UNLIKELY( scratch_top > (ulong)scratch + scratch_footprint( tile ) ) )
     FD_LOG_ERR(( "scratch overflow %lu %lu %lu", scratch_top - (ulong)scratch - scratch_footprint( tile ), scratch_top, (ulong)scratch + scratch_footprint( tile ) ));
+
+  ctx->ip = init_ctx->ip;
```

Comment:
Might be better to move this line up a bit, with related code, but at least it shouldn't be after the scratch overflow check which should be last.

Reply:
The scratch overflow check is last in the function so that whatever gets added, it checks the right thing. You could refactor the function to be like
```
init_scratch()
do_work()
check_scratch()
```
This `ctx->ip` would be part of `do_work` so it should go above the scratch check.

Reply:
ctx is in the memory region allocated. You want to modify the region before checking the region is valid?
I'll make it consistent